src/report/generate_trustymail_report.py

```diff
@@ -0,0 +1,1372 @@
+        result["agency_id"] = self.__agency_id
+        result["spf_covered_count"] = self.__spf_covered_count
+        result["spf_coverered_percentage"] = self.__spf_coverered_percentage
+        result["has_no_weak_crypto_count"] = self.__has_no_weak_crypto_count
+        result["has_no_weak_crypto_percentage"] = self.__has_no_weak_crypto_percentage
+        result["valid_dmarc_count"] = self.__valid_dmarc_count
+        result["valid_dmarc_percentage"] = self.__valid_dmarc_percentage
+        result["valid_dmarc_reject_count"] = self.__valid_dmarc_policy_of_reject_count
+        result["valid_dmarc_reject_percentage"] = self.__valid_dmarc_reject_percentage
+        result[
+            "valid_dmarc_bod1801_rua_uri_count"
+        ] = self.__valid_dmarc_bod1801_rua_uri_count
+        result[
+            "valid_dmarc_bod1801_rua_uri_percentage"
+        ] = self.__valid_dmarc_bod1801_rua_uri_percentage
+        result["domain_supports_smtp_count"] = self.__domain_supports_smtp_count
+        result[
+            "base_domain_supports_smtp_count"
+        ] = self.__base_domain_supports_smtp_count
+        result["subdomain_supports_smtp_count"] = (
+            self.__domain_supports_smtp_count - self.__base_domain_supports_smtp_count
+        )
+        result[
+            "base_domain_plus_smtp_subdomain_count"
+        ] = self.__base_domain_plus_smtp_subdomain_count
+        result["supports_starttls_count"] = self.__supports_starttls_count
+        result["supports_starttls_percentage"] = self.__supports_starttls_percentage
+        result["bod_1801_compliant_count"] = self.__bod_1801_compliant_count
+        result["bod_1801_compliant_percentage"] = self.__bod_1801_compliant_percentage
+
+        self.__latex_escape_structure(result["report_doc"])
+
+        with open(filename, "w") as out:
+            out.write(json.dumps(result))
+
+    def __generate_latex(self, mustache_file, json_file, latex_file):
+        template = codecs.open(mustache_file, "r", encoding="utf-8").read()
+
+        with codecs.open(json_file, "r", encoding="utf-8") as data_file:
+            data = json.load(data_file)
+
+        r = pystache.render(template, data)
+        with codecs.open(latex_file, "w", encoding="utf-8") as output:
+            output.write(r)
+
+    def __generate_final_pdf(self):
+        xelatex = ["/usr/bin/xelatex", REPORT_TEX]
+        # Bandit frowns upon the use of subprocess, but we need it
+        # here.  Hence the nosec.
+        subprocess.run(xelatex)  # nosec B603
+        subprocess.run(xelatex)  # nosec B603
+
+
+def main():
+    """Create Trustworthy Email Agency Report PDF."""
+    args = docopt(__doc__, version="v0.0.1")
+
+    # Set up logging
+    logging.basicConfig(
+        format="%(asctime)-15s %(levelname)s %(message)s", level=logging.INFO
+    )
+
+    db = db_from_config(DB_CONFIG_FILE)
+
+    print("Generating Trustymail Report...")
+    # TODO: Use agency ID instead of full agency name
```

Comment:
Reminder to create an issue and mention it in a comment here.

Reply:
I don't know why this comment is there, or why the ID would be preferred here.  Can I just delete this comment?

Reply:
I think we have generally preferred to use the CyHy org ID with all of our scripts whenever possible.  I'm guessing that back when this script was first created, there was no easy way to get the CyHy org IDs, but I think that is no longer the case.

Reply:
See commit bc295a3.

Reply:
Thanks!